### ISSUE TITLE
Use Artifact system to ship test .root files

### ISSUE THREAD

**Issue by Moelf** [JuliaHEP/UnROOT.jl#173]
https://github.com/simeonschaub/ArtifactUtils.jl

**Comment by Yuan-Ru-Lin** [comment]
Where can I download the file (as a `.tar.gz`)?

**Comment by Yuan-Ru-Lin** [comment]
Or maybe the question should be "where should the file be uploaded to?"

**Comment by Moelf** [comment]
yeah right now they are https://github.com/JuliaHEP/UnROOT.jl/tree/master/test/samples

we can upload to a repo in JuliaHEP, and then you can point to URLs like: https://pkgdocs.julialang.org/v1/artifacts/#Basic-Usage

**Comment by Yuan-Ru-Lin** [comment]
Makes sense. Can I be a member of JuliaHEP and do that?

**Comment by tamasgal** [comment]
Yep sure, I invited you 🙂 We can figure out the details of that repository later.

**Comment by Yuan-Ru-Lin** [comment]
According to https://pkgdocs.julialang.org/v1/artifacts/, it seems that we need to compress those sample files before we can list them as Artifacts. Should they be compressed separately, or pack them into a single `.tar.gz`? What do you think?

**Comment by Moelf** [comment]
I believe they have to be `tarball` (`.tar`), but doesn't have to be compressed (`.tar.gz`). But yeah I think pack everything into a single tarball is fine.

**Comment by Yuan-Ru-Lin** [comment]
I created a `.tar` file by `tar -xf samples.tar samples/` and there was this error message

```
- file:///Users/yuanrulin/Projects/turn-sample-files-into-artifacts/samples.tar
    Error: This does not appear to be a TAR file/stream — malformed chksum field: ".root". Note: Tar.jl does not handle decompression; if the tarball is compressed you must use an external command like `gzcat` or package like CodecZlib.jl to decompress it. See the README file for examples.
```

If I use `.tar.gz` instead, things are good. Maybe we should go with `.tar.gz`?

**Comment by Yuan-Ru-Lin** [comment]
But I do realize that the artifact will have to change if there are new files in the future. Maybe that's something of concern, or maybe not?

**Comment by Moelf** [comment]
good point... maybe we should keep these small files as how they are for now, and if we have larger (>5MB) test files in the future, use artifact system for those one by one to avoid changing artifact every time?

**Comment by Yuan-Ru-Lin** [comment]
Yeah. That sounds reasonable. I'd say maybe those test samples should be grouped according to some criteria. Otherwise, we will have to write something like `joinpath(artifact"km3net_online", "km3net_online.root")` which looks cumbersome. In short, artifacts are more like directories, instead of files.

**Comment by Yuan-Ru-Lin** [comment]
Another possibility is to make a `.tar.gz` for every sample file. and then define a macro such as

```julia
macro samples_str(file)
    :( joinpath(@artifact_str(replace($file, ".root"=>"")), $file) )
end
```

so that `sample"km3_online.root"` will result in `joinpath(artifact"km3net_online", "kme3net_online.root")`.

What do you think?
